Game.cpp

```diff
@@ -0,0 +1,146 @@
+#include "Game.h"
+#include "Map.h"
+
+void Game::setMap(Map map)
+{
+    
+}
+void Game::putHero(Hero hero,int x,int y)
+{
+    if(map.get(x,y) == Map::Free)
+    {
+        hero_location = std::make_pair(x+1,y);
+        this->hero = new Hero(hero);
+    }
+    else throw Game::OccupiedException("Location is not avaibile! \n");
+}
+void Game::putMonster(Monster monster,int x, int y)
+{
+    if(map.get(x,y) == Map::Free && Game::checkForMonsters(x,y) < 2)
+        monster_locations.push_back(std::make_pair(monster,std::make_pair(x,y)));
+
+    else throw Game::OccupiedException("Location is not avaibile! \n");
+}
+bool Game::freetoStep(int x,int y) const
+{
+    for(auto iter = monster_locations.begin();iter != monster_locations.end();iter++)
+    {
+        if(iter->second.first == x && iter->second.second == y)
+            return false;
+    }
+    if((hero_location.first == x && hero_location.second == y) || (hero_location.first-1 == x && hero_location.second == y))
+        return false;
+    
+    return true;
+}
+void Game::stepOn(int x, int y)
+{
+    if(map.get(x-1,y) == Map::Wall) throw Game::OccupiedException("Can't move here there is a Wall !\n");
+
+    for(auto iter = monster_locations.begin();iter != monster_locations.end();iter++)
+    {
+        if(iter->second.first == x - 1 && iter->second.second == y)
+        {
+            this->hero->fightTilDeath(iter->first);
+            if(hero->isAlive())
+            {
+                iter = monster_locations.erase(iter);
+            }
+        }
+        if(iter->second.first +1 == x && iter->second.second == y)
+        {
+            this->hero->fightTilDeath(iter->first);
+            if(hero->isAlive())
+            {
+                iter = monster_locations.erase(iter);
+            }
+        }
+    }
+    hero_location = std::make_pair(x,y);
+}
+void Game::run()

```

Comment:
ebből is hiányoznak a vizsgálatok; van-e map beállítva, van-e hero beállítva

Reply:
done